cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,337 @@
+	para: IdentityInfoParachain,
+}
+
+fn identities() -> Vec<Identity> {
+	let pgp_fingerprint = [
+		0x12, 0x34, 0x56, 0x78, 0x9A, 0xBC, 0xDE, 0xF0, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
+		0x88, 0x99, 0xAA, 0xBB, 0xCC,
+	];
+	// Minimal Identity
+	let first_relay = IdentityInfo {
+		display: Data::None,
+		legal: Data::None,
+		web: Data::None,
+		email: Data::None,
+		pgp_fingerprint: None,
+		image: Data::None,
+		twitter: Data::None,
+		riot: Data::None,
+		additional: Default::default(),
+	};
+	let first_para = IdentityInfoParachain {
+		display: Data::None,
+		legal: Data::None,
+		web: Data::None,
+		matrix: Data::None,
+		email: Data::None,
+		pgp_fingerprint: None,
+		image: Data::None,
+		twitter: Data::None,
+		github: Data::None,
+		discord: Data::None,
+	};
+	// Full Identity with no additional
+	let second_relay = IdentityInfo {
+		display: Data::Raw(b"xcm-test-one".to_vec().try_into().unwrap()),
+		legal: Data::Raw(b"The Right Ordinal Xcm Test, Esq.".to_vec().try_into().unwrap()),
+		web: Data::Raw(b"https://xcm-test.io".to_vec().try_into().unwrap()),
+		email: Data::Raw(b"xcm-test@gmail.com".to_vec().try_into().unwrap()),
+		pgp_fingerprint: Some(pgp_fingerprint),
+		image: Data::Raw(b"xcm-test.png".to_vec().try_into().unwrap()),
+		twitter: Data::Raw(b"@xcm-test".to_vec().try_into().unwrap()),
+		riot: Data::Raw(b"riot-xcm-test".to_vec().try_into().unwrap()),
+		additional: Default::default(),
+	};
+	let second_para = IdentityInfoParachain {
+		display: Data::Raw(b"xcm-test-one".to_vec().try_into().unwrap()),
+		legal: Data::Raw(b"The Right Ordinal Xcm Test, Esq.".to_vec().try_into().unwrap()),
+		web: Data::Raw(b"https://xcm-test.io".to_vec().try_into().unwrap()),
+		matrix: Data::Raw(b"riot-xcm-test".to_vec().try_into().unwrap()),
+		email: Data::Raw(b"xcm-test@gmail.com".to_vec().try_into().unwrap()),
+		pgp_fingerprint: Some(pgp_fingerprint),
+		image: Data::Raw(b"xcm-test.png".to_vec().try_into().unwrap()),
+		twitter: Data::Raw(b"@xcm-test".to_vec().try_into().unwrap()),
+		github: Data::None,
+		discord: Data::None,
+	};
+	// Full Identity with nonsensical additional
+	let mut third_relay = second_relay.clone();
+	third_relay.additional = BoundedVec::try_from(vec![(
+		Data::Raw(b"foO".to_vec().try_into().unwrap()),
+		Data::Raw(b"bAr".to_vec().try_into().unwrap()),
+	)])
+	.unwrap();
+	// Full Identity with meaningful additional
+	let mut fourth_relay = second_relay.clone();
+	fourth_relay.additional = BoundedVec::try_from(vec![
+		(
+			Data::Raw(b"github".to_vec().try_into().unwrap()),
+			Data::Raw(b"niels-username".to_vec().try_into().unwrap()),
+		),
+		(
+			Data::Raw(b"discord".to_vec().try_into().unwrap()),
+			Data::Raw(b"bohr-username".to_vec().try_into().unwrap()),
+		),
+	])
+	.unwrap();
+	let mut fourth_para = second_para.clone();
+	fourth_para.github = Data::Raw(b"niels-username".to_vec().try_into().unwrap());
+	fourth_para.discord = Data::Raw(b"bohr-username".to_vec().try_into().unwrap());
```

Comment:
There are only minor differences between the four cases, and in most cases no difference between para and relay identity. Since you've already defined the `Identity` struct I think a constructor would tidy this code up a lot and reduce the repetition. Just something as simple as `new(full: bool, additional: Option<(_,_)>)` would cover it.

Instead of copying and pasting the code here with slightly different mock data each time since it's largely irrelevant to this test you could just generalise it in the constructor and have a `let mock_data = if full { Data::Raw(..) } else { Data::None };` and clone mock_data into each field except `additional, github, discord` which depend on the second argument.

Reply:
This was good as it removed the need for me to use a Vec to initialise the tests, I can just `Identity::new()`, passing in the different args for each case.